utils/ixdtf/src/parsers/time.rs

```diff
@@ -0,0 +1,212 @@
+    };
+
+    cursor.close()?;
+
+    Ok(Some(IsoParseRecord {
+        date: None,
+        time: Some(time),
+        offset: utc_offset,
+        tz,
+        calendar: annotations.calendar,
+        annotations: Vec::default(),
+    }))
+}
+
+/// Parse `TimeRecord`
+pub(crate) fn parse_time_record(cursor: &mut Cursor) -> ParserResult<TimeRecord> {
+    let hour = parse_hour(cursor)?;
+
+    if !cursor.check_or(false, |ch| is_time_separator(ch) || ch.is_ascii_digit()) {
+        return Ok(TimeRecord {
+            hour,
+            minute: 0,
+            second: 0,
+            millisecond: 0,
+            microsecond: 0,
+            nanosecond: 0,
+        });
+    }
+
+    let separator_present = cursor.check_or(false, is_time_separator);
+    cursor.advance_if(separator_present);
+
+    let minute = parse_minute_second(cursor, false)?;
+
+    if !cursor.check_or(false, |ch| is_time_separator(ch) || ch.is_ascii_digit()) {
+        return Ok(TimeRecord {
+            hour,
+            minute,
+            second: 0,
+            millisecond: 0,
+            microsecond: 0,
+            nanosecond: 0,
+        });
+    }
+
+    let second_separator = cursor.check_or(false, is_time_separator);
+    assert_syntax!(separator_present == second_separator, TimeSeparator);
+    cursor.advance_if(second_separator);
+
+    let second = parse_minute_second(cursor, true)?;
+
+    let (millisecond, microsecond, nanosecond) = parse_fraction_component(cursor)?;
+
+    Ok(TimeRecord {
+        hour,
+        minute,
+        second,
+        millisecond,
+        microsecond,
+        nanosecond,
+    })
+}
+
+/// Parse an hour value.
+#[inline]
+pub(crate) fn parse_hour(cursor: &mut Cursor) -> ParserResult<u8> {
+    let first = cursor.next_digit()?.ok_or(ParserError::TimeHour)?;
+    let hour_value = first * 10 + cursor.next_digit()?.ok_or(ParserError::TimeHour)?;
+    if !(0..=23).contains(&hour_value) {
+        return Err(ParserError::TimeHour);
+    }
+    Ok(hour_value)
+}
+
+/// Parses `MinuteSecond` value.
+#[inline]
+pub(crate) fn parse_minute_second(cursor: &mut Cursor, is_second: bool) -> ParserResult<u8> {
+    let (valid_range, err) = if is_second {
+        (0..=60, ParserError::TimeSecond)
+    } else {
+        (0..=59, ParserError::TimeMinute)
+    };
+    let first = cursor.next_digit()?.ok_or(err)?;
+    let min_sec_value = first * 10 + cursor.next_digit()?.ok_or(err)?;
+    if !valid_range.contains(&min_sec_value) {
+        return Err(err);
+    }
+    Ok(min_sec_value)
+}
+
+#[inline]
+pub(crate) fn parse_fraction_component(cursor: &mut Cursor) -> ParserResult<(u16, u16, u16)> {
+    let Some(fraction) = parse_fraction(cursor)? else {
+        return Ok((0, 0, 0));
+    };
+    Ok(to_fraction_components(fraction))
+}
+
+/// Parse a `Fraction` value
+///
+/// This is primarily used in ISO8601 to add percision past
+/// a second.
+#[inline]
+pub(crate) fn parse_fraction(cursor: &mut Cursor) -> ParserResult<Option<f64>> {
```

Comment:
Suggestion: I think `f64` is not the right choice here since it might be lossy; a `u32` of nanoseconds should be a better choice.